proactive-healthcheck/api-health-check.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+#set -x
+
+#input parameters
+#username to connect to the ProActive Portals
+username=changeme
+#password of the user connecting to the ProActive Portals
+proactiveAdminPasswordDecrypted=changeme
+#server DNS name or public IP
+serverDNSName=try.activeeon.com
+#server Port
+webPort=8443
+#web protocol, http or https
+webProtocol=https
+#name of one of the node sources, needed in a test to get its description
+nsName=local
+
+#set colors for printf state of services
+RED='\033[0;31m'
+NC='\033[0m' # No Color
+GREEN='\033[0;32m'
+PURPLE='\033[0;35m'
+
+#number of failed tests
+FAIL=0
+#number of successful tests
+SUCCESS=0
+
+#complete ProActive server web URL
+URL="${webProtocol}://${serverDNSName}:${webPort}"
+
+#function used by tests to check the returned status and print messages
+check_endpoint() {
+  endpoint=$1
+  message=$2
+  if [ "$endpoint" == "200" ]; then
+    echo -e "${GREEN} OK ${message}"
+    ((SUCCESS++))
+  else
+    echo -e "${RED} NOK ${message}"
+    ((FAIL++))
+  fi
+}
+
+#get the session id, needed for all tests
+endpoint=$(curl -o - -d "username=${username}&password=${proactiveAdminPasswordDecrypted}" --write-out "\n%{http_code}\n" --silent --output /dev/null ${URL}/rest/scheduler/login)
+response=($endpoint)
+sessionid=${response[0]}
+status=${response[1]}
+message="$NC: ${PURPLE}(POST: scheduler/login)$NC GET A SESSION ID"
+if [ "$status" == "200" ]; then
+  ((SUCCESS++))
+  echo -e "${GREEN} OK $message"
+else
+  echo -e "${RED} NOK $message"
+  ((FAIL++))
+  #exit the script if the session id could not be retrieved since it is required by most of the tests
+  exit 1
+fi
+
+#wait until a node becomes free before starting the tests
+nodeState=""
+until [[ $nodeState == "FREE" ]]; do
+  monitoring=$(curl -G --header "sessionid:$sessionid" --silent "$URL/rest/rm/monitoring/")
+  nodeEvent=$(echo $monitoring | jq -r '.nodesEvents[0]')
+  nodeState=$(echo $nodeEvent | jq -r '.nodeState')
+done
```

Comment:
What if the node never becomes available?
I deem we need to add a number of attempts and a delay after each attempt, if all the attempts fail, we declare the deployment as failed.

Reply:
I took into account your remark. Please check it again. Thanks.
NB: I have also added the total elapsed time of script execution.